ifrs17/Utils/ActivityLogsAndImports.ipynb

```diff
@@ -0,0 +1,302 @@
+{
+    "metadata": {
+        "authors": [],
+        "kernelspec": {
+            "display_name": "Formula Framework",
+            "language": "C#",
+            "name": "C#"
+        },
+        "language_info": {
+            "file_extension": ".cs",
+            "mimetype": "text/plain",
+            "name": "C#"
+        }
+    },
+    "nbformat": 4,
+    "nbformat_minor": 5,
+    "cells": [
+        {
+            "cell_type": "code",
+            "source": [
+                "#!import \"../DataModel/DataStructure\""
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "using System.IO;",
+                "\nusing System.Text;",
+                "\nusing Systemorph.Vertex.DataSetReader.Csv;",
+                "\nusing Systemorph.Vertex.Session;",
+                "\nusing Systemorph.Vertex.Import.Builders;",
+                "\nusing Systemorph.Vertex.Import.Mappings;",
+                "\nusing Systemorph.Vertex.DataSetReader;",
+                "\n",
+                "\npublic record StreamWrapper(Stream Stream, bool WillBeReused);",
+                "\n",
+                "\npublic static string ProcessNotification(this object obj) => obj is ActivityMessageNotification amn ? amn.Message : \"\"; "
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record KeyedActivity : KeyedRecord",
+                "\n{",
+                "\n    public string Username {get; init;}",
+                "\n",
+                "\n    public DateTime StartDateTime {get; init;}",
+                "\n",
+                "\n    public DateTime EndDateTime {get; init;}",
+                "\n",
+                "\n    public ActivityLogStatus Status {get; init;}",
+                "\n",
+                "\n    public string Category {get; init;}",
+                "\n",
+                "\n    [Conversion(typeof(JsonConverter<string[]>))]",
+                "\n    public string[] ErrorMessages {get; init;}",
+                "\n",
+                "\n    [Conversion(typeof(JsonConverter<string[]>))]",
+                "\n    public string[] WarningMessages {get; init;}",
+                "\n",
+                "\n    [Conversion(typeof(JsonConverter<string[]>))]",
+                "\n    public string[] InfoMessages {get; init;}",
+                "\n",
+                "\n    public Guid SourceId {get; init;} // Check if converting to Guid causes issues. If not, leave Guid.",
```

Comment:
Is this check in scope, or a future TODO?

Reply:
Done now :)